apis/exoscale/v1/conditions.go

```diff
@@ -0,0 +1,48 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// The reason we need a custom condition type is that metav1.Condition declares all fields as required.
+// However, we have seen issues where Crossplane's properties in Conditions aren't all required and lead to Crossplane not being able to copy conditions.
+
+type Condition struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Pattern=`^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])$`
+	// +kubebuilder:validation:MaxLength=316
+
+	// Type of condition.
+	Type string `json:"type,omitempty"`
+
```

Comment:
This file is a copy of the one in `apis/v1` right?  Do we need to duplicate that or could we just import it from `apis/v1`?

Reply:
Fixed as well.